fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/cpp/utils/Attributes.xtend

```diff
@@ -205,6 +205,19 @@ class Attributes {
 		}
 	'''
 
+	//======================================================
+	// Define constants for attributes
+	//======================================================
+	def attributeConstant(Attribute attribute) '''
+		«IF attribute.isSpectrum»
+			// Constants for «attribute.name» attribute
+			static const long «attribute.name»_X_DATA_SIZE=«attribute.maxX»
+		«ELSEIF attribute.isImage»
+			// Constants for «attribute.name» attribute
+			static const long «attribute.name»_X_DATA_SIZE=«attribute.maxX»
+			static const long «attribute.name»_Y_DATA_SIZE=«attribute.maxY»
+		«ENDIF»
```

Comment:
1. This does not compile, comma is missing at the end of declaration,
2. I think it would be good to declare these constants in some TU. Someone may odr-use them and linking will fail (or there can be no diagnostic at all, see: https://stackoverflow.com/questions/33812659/basic-fun-with-static-const-integral-members). With clang 11.0.1 with -O0 I'm getting following error:
   ```
   /usr/bin/ld: /tmp/Mytestclass-effff5.o: in function `Mytestclass_ns::Mytestclass::read_myimg(Tango::Attribute&)':                                                             
   Mytestclass.cpp:(.text+0x15cf): undefined reference to `Mytestclass_ns::myimgAttrib::myimg_X_DATA_SIZE'  
   ```
   From following code:
   ```cpp
   long x = *&myimgAttrib::myimg_X_DATA_SIZE;
   ```
3. Even better would be to make them `constexpr`. This would solve above ODR problems and will allow using these constants in contexts where a compile-time constant expression is required. I'm not sure which C++ standard release Pogo-generated code needs to support, but in cppTango (tango-9-lts, upcoming 9.4) we currently require C++14 so `constexpr` is safe in that case.
4. I suggest to put a space around the assignment operator for consistency with other generated code.

Reply:
One more comment. Point 3 where I said that `constexpr` provides definition is not entirely true. This is the case only with c++17 or better. For c++14 we still need to provide definition somewhere. See below link for standard excerpts and a more realistic example than just applying `*&` to these constants:
https://stackoverflow.com/questions/52251833/is-static-constexpr-variable-odr-used

Reply:
Yes it's ok now IMO. Just add space between type and name because it is not compiling. Also remove static because this is another compilation error. I strongly suggest to test compilation locally before pushing non-trivial changes.